src/main/java/Storage.java

```diff
@@ -0,0 +1,200 @@
+     * @param type Todo, Event or Deadline.
+     * @param time The date and time for the Tasks.
+     * @throws IOException Throws exception.
+     */
+    public void convertToHardDiskFormatAndStore(Task task, String type, String time) throws IOException {
+
+        String doneOrNotDone = "";
+        if (task.isDone) {
+            doneOrNotDone += "O";
+        } else {
+            doneOrNotDone += "X";
+        }
+
+        String newDescription = "";
+        if (type.equals("T")) {
+            //todo
+            newDescription += type + " | " + doneOrNotDone + " | " + task.description;
+        } else {
+            //event & deadline
+            newDescription += type + " | " + doneOrNotDone + " | " + task.description + " | " + time;
+        }
+
+        File file = new File(filePath);
+        FileWriter fr = new FileWriter(file, true);
+
+        //System.out.println("size of list is: " + storingList.size());
+        if (file.length() == 0) {
+            fr.write(newDescription);
+        } else {
+            fr.write("\n" + newDescription);
+        }
+
+        //System.out.println("size of list is: " + storingList.size());
+        //Scanner myReader = new Scanner(file);
+        fr.close();
+    }
+
+    /**
+     * Modify the done value in the dukeStorage.txt file.
+     * Change not done (X) into done (O)
+     *
+     * @param lineNumber The line number to be deleted from hard disk.
+     * @throws IOException Throws exception.
+     */
+    public void replaceLineFromHardDisk(int lineNumber) throws IOException {
+        File myObj = new File(filePath);
+        Scanner myReader = new Scanner(myObj);
+
+        String newData = "";
+        int counter = 1;
+        while (myReader.hasNextLine()) {
+            String data = myReader.nextLine();
+            //System.out.println(data);
+
+            if (counter == lineNumber) {
+                data = data.substring(0, 4) + "O" + data.substring(5, data.length());
+            }
+
+            if (counter == 1) {
+                newData += data;
+            } else {
+                newData += "\n" + data;
+            }
+
+            counter++;
+        }
+
+        FileOutputStream fileOut = new FileOutputStream(filePath);
+        fileOut.write(newData.getBytes());
+        fileOut.close();
+    }
+
+    /**
+     * Delete the corresponding line from dukeStorage.txt file according to the task deleted.
+     *
+     * @param lineNumber The line number to be deleted from hard disk.
+     * @throws IOException Throws exception.
+     */
+    public void deleteLineFromHardDisk(int lineNumber) throws IOException {
+        File myObj = new File(filePath);
+        Scanner myReader = new Scanner(myObj);
+
+        String newData = "";
+        boolean firstLineDone = false;
+        int counter = 1;
+
+        while (myReader.hasNextLine()) {
+            String data = myReader.nextLine();
+
+            if (counter == lineNumber) {
+                counter++;
```

Comment:
You could place counter++ outside the if loop so that you don't have to repeat it.